src/main/java/com/assignment1/parser/CommandLineParser.java

```diff
@@ -0,0 +1,138 @@
+package com.assignment1.parser;
+
+import com.assignment1.Type;
+import com.assignment1.exceptions.GenericApplicationException;
+import com.assignment1.item.ItemDTO;
+import com.assignment1.item.ItemEntity;
+import java.math.BigDecimal;
+
+/**
+ * Class to implement Parser.
+ */
+public class CommandLineParser implements Parsable {
+
+  /**
+   * Object of Valid class to be used in Parse method.
+   */
+  private final transient Valid validator;
+
+  /**
+   * Constructor to hardcode the Valid object.
+   */
+  public CommandLineParser() {
+    validator = new Validator();
+  }
+
+  /**
+   * Constructor to initialize Valid object using parameter.
+   */
+  public CommandLineParser(final Valid validator) {
+    this.validator = validator;
+  }
+
+  /**
+   * Explicit check for name option because it contains multiple words.
+   * Return count of words or throws exception if name not present (blank).
+   */
+  public int findNumOfWordsInName(final String... args) throws GenericApplicationException {
+    int numWords = 0;
+    if ("-name".equals(args[0])) { //First argument mandatory to be name
+      int ptr = 1; //pointer to check for how many words name is extended to.
+      while (ptr < args.length && args[ptr].charAt(0) != '-') {
+        //loop until we get option with first character '-'
+        numWords += 1; //Increase count for significant arguments
+        ptr++;
+      }
+    } else {
+      throw new GenericApplicationException("Invalid Input: -name option not present at first place");
+    }
+    return numWords;
+  }
+
+  /**
+   * function to extract name.
+   */
+  private String extractName(final int numOfWordsInName, final String... args) {
+    final StringBuilder stringBuilder = new StringBuilder();
+    for (int i = 1; i <= numOfWordsInName - 1; i++) {  //Building up the full name of item
+      stringBuilder.append(args[i]).append(" ");
+    }
+    stringBuilder.append(args[numOfWordsInName]); //Putting last word separately to avoid space
+    return stringBuilder.toString();
+  }
+
+  /**
+   * Building Item entity when everything is right and valid.
+   */
+  private ItemEntity buildUpItemEntity(final ItemDTO itemDTO) {
+    final ItemEntity item = new ItemEntity();
+    item.setName(itemDTO.getName());
+    if ("".equals(itemDTO.getPrice())) {
+      item.setPrice(BigDecimal.valueOf(0.0));
+    } else {
+      item.setPrice(
+              new BigDecimal(itemDTO.getPrice().replace(",","")));
+    }
+    if ("".equals(itemDTO.getQuantity())) {
```

Comment:
use this instead StringUtls.isEmpty()

Reply:
Used .isEmpty() function wherever needed.